### PR TITLE
Improve node selector details

### PR DESCRIPTION
**Pull request by MohamedElmdary**
### Description
Improve node selector details

### Changes
- redesign node card
- write a selectable table (custom component) to work as VSelect

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1872
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1870

## SS
1. Initial Loading (loading while there is no loaded nodes yet)
image

2. After first loading pick a valid node and scroll down to that node (to it's center) smoothly
image

3. Show load more nodes (if there is more nodes to load)
image

4. Ask to reload node if data changed
image

5. Show error if the provided filters is not valid
image

6. Show message while validating
image

7. Show loading linear line (if loading and there is nodes already there)
image

8. Got hint as before
image

9. Has hint to notify user that cpu might be more than 100% (usage)
image

10. some more tooltips for more details
image
image

11. Deploy button disabled
image
image

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by AlaaElattar** [comment]
i think the separator between cards should be stronger to be clear. what do u think ?

**Comment by MohamedElmdary** [comment]
> i think the separator between cards should be stronger to be clear. what do u think ?

VDivider style doesn't match with outer border so I added a border bottom instead

**Review by xmonader — CHANGES_REQUESTED** [review]
Let's not show a serial if it's default string

**Review by AhmedHanafy725 — COMMENTED** [review]
what does it mean the shared chip on the farm? I prefer to have farm name instead.

**Review by mohamedamer453 — COMMENTED** [review]
- After the nodes are loaded it randomly scrolls to one of the nodes, i think it should load the nodes and leave it up to the user to select the desired node.

[Screencast from 08-01-24 15:54:16.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/d35033ab-e0e5-452d-8996-aafac77be7b6)

- I believe as it was agreed upon before that the user doesn't need to know anything about the term storage pools and it would be better if the message was updated to something else like "checking if the deployment will fit in the node's disks" or something similar.

  image


- IMO it would look better if the node id was written in the same way as the farm id. e.g. "Node ID: 11" instead of using brackets.

   image

- In the manual filter, when the dedicated node filter is enabled and a shared node id is added it should mention that this node is shared in the error message instead of mentioning that this node is not rented.

   image

- On mainnet, some nodes appeared without flag. e.g. node 2336

 image

even though the node has a country in its details from the gridproxy

image

- After Loading more nodes the storage pools check is done even when the same node is selected.

   [Screencast from 08-01-24 16:40:48.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/101194226/af3951f8-a3b2-400f-b590-c5a6769a1fdd)

**Comment by MohamedElmdary** [comment]
> what does it mean the shared chip on the farm? I prefer to have farm name instead.

For farmName we need extra request

**Comment by MohamedElmdary** [comment]
> * After the nodes are loaded it randomly scrolls to one of the nodes, i think it should load the nodes and leave it up to the user to select the desired node.

It was a requested feature to improve ux

**Comment by mohamedamer453** [comment]
> * After Loading more nodes the storage pools check is done even when the same node is selected.
>   

This issue is still present

**Comment by MohamedElmdary** [comment]
> > * After Loading more nodes the storage pools check is done even when the same node is selected.
> 
> This issue is still present


It's verify that node still valid

File: packages/playground/src/components/node_selector/TfAutoNodeSelector.vue [hunk 1]
```diff
@@ -1,93 +1,127 @@
...
+                      ? nodeInputValidateTask.loading
+                        ? 'Pending'
+                        : nodeInputValidateTask.data
+                        ? 'Valid'
+                        : 'Invalid'
+                      : 'Init'
+                  "
+                />
+              </div>
+              <!-- <div class="border-b" :style="{ borderBottomWidth: '2px !important' }" /> -->
```
Comment (mohamedamer453) [inline packages/playground/src/components/node_selector/TfAutoNodeSelector.vue:77]: remove the commented line

**Review by mohamedamer453 — APPROVED** [review]

**Review by 0oM4R — APPROVED** [review]

**Review by xmonader — APPROVED** [review]

**Pull request merged**